### PR TITLE
[JENKINS-68809] Topic repository listing should include forks by default

### PR DESCRIPTION
**Pull request by Dohbedoh**
# Description

[JENKINS-68809](https://issues.jenkins.io/browse/JENKINS-68809): the recent Topic filter added does not include fork by default. Due to the default behavior of the Search API. It probably should given that there is now an `ExcludeForkedRepositoriesTrait`.

This PR add `fork:true` to the search unless `ExcludeForkedRepositoriesTrait` is added.

Note: we could actually always add `fork:true` and let the API filter out the results. But by using the filter in the search query,  we can ave on that computation.

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Automated tests have been added to exercise the changes
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

@jenkinsci/github-branch-source-plugin-developers

### CODE REVIEW

**Comment by rsandell** [comment]
@Dohbedoh To you see this as a bugfix or a feature/enhancement? For me it's a bit of a gray area :)

**Comment by Dohbedoh** [comment]
I was thinking of it as a bug given the facts that the event mechanism considers forks, which is inconsistent

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java [hunk 1]
```diff
@@ -1247,17 +1259,14 @@ public void visitSources(SCMSourceObserver observer) throws IOException, Interru
...
-            GitHubConsoleNote.create(
-                System.currentTimeMillis(),
-                String.format("Looking up repositories for topics: '%s'", topics)));
+  private Iterable<GHRepository> searchRepositories(
+      final GitHub github, final GitHubSCMNavigatorContext context) {
     final GHRepositorySearchBuilder ghRepositorySearchBuilder = github.searchRepositories();
-    topics.forEach(ghRepositorySearchBuilder::topic);
-    ghRepositorySearchBuilder.q("org:" + getRepoOwner());
+    context.getTopics().forEach(ghRepositorySearchBuilder::topic);
+    ghRepositorySearchBuilder.org(getRepoOwner());
```
Comment (jetersen) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java:1266]: Thanks for updating this :)

I had put a pull request into github api a while ago

https://github.com/hub4j/github-api/pull/1355

**Comment by joshbranham** [comment]
Looks like this resolves https://issues.jenkins.io/browse/JENKINS-68765 thank you!

**Pull request merged**